source/common/router/config_impl.cc

```diff
@@ -490,11 +492,11 @@ const VirtualHostImpl* RouteMatcher::findVirtualHost(const Http::HeaderMap& head
   auto iter = virtual_hosts_.find(headers.Host()->value().c_str());
   if (iter != virtual_hosts_.end()) {
     return iter->second.get();
-  } else if (default_virtual_host_) {
-    return default_virtual_host_.get();
+  } else if (VirtualHostPtr vhost =
+                 *wildcard_virtual_host_suffixes_.find(headers.Host()->value().c_str()).first) {
+    return vhost.get();
   }
-
-  return nullptr;
+  return default_virtual_host_.get();
```

Comment:
if (uniq) { return uniq.get() } else { return nullptr; } is the same as return uniq.get().